src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java

```diff
@@ -144,9 +143,17 @@ private void refreshTextFlow() {
     }
 
     private Text buildText(String content, TextDecoration decoration ) {
-        Text text = new Text(content);
-        Optional.ofNullable(decoration.getForeground()).ifPresent(text::setFill); // has to be fill for font to render properly
-        Optional.ofNullable(decoration.getFont()).ifPresentOrElse(text::setFont, () -> text.setFont(TextDecoration.DEFAULT.getFont()) );
+        Objects.requireNonNull(decoration);
+        Text text = new Text(Objects.requireNonNull(content));
+        text.setFill(decoration.getForeground());
+
+        // Cashing fonts, assuming their, especially for default one
+        Map<String, Font> fontCache = new HashMap<>();
```

Comment:
This is being created on every refresh, isn't it? Move it to a global field

Reply:
Not sure it is a good idea though - the component is going to keep those font refs all the time if we do.
Just wanted to speedup refreshes a little.

Reply:
I don't see an issue with having a global cache based on fonts created out of their attributes. That will be faster than creating the cache and the used fonts on every refresh.

If decorations change later on, these font attributes will change too, and the cache will have to create such new fonts. 

The only issue I see then is that we wouldn't clean it up if fonts are not longer used.

Reply:
That is doable, if we add expiration to that cache

Reply:
Moved font cache to the class level, also added eviction capability